spyder/plugins/onlinehelp/widgets.py

```diff
@@ -78,123 +101,384 @@ def completer(self):
+                # assign the same actions to the children widgets in order
+                # for shortcuts to work
+                self.webview.addAction(action)
+
+        self.initialize()
+
+    def update_actions(self):
+        stop_action = self.get_action(WebViewActions.Stop)
+        refresh_action = self.get_action(WebViewActions.Refresh)
+
+        refresh_action.setVisible(not self._is_running)
+        stop_action.setVisible(self._is_running)
+
+    def on_option_update(self, option, value):
+        pass
 
-    def initialize_continued(self):
-        """Load home page"""
+    # --- Private API
+    # ------------------------------------------------------------------------
+    def _start(self):
+        """Webview load started."""
+        self._is_running = True
+        self.start_spinner()
+        self.update_actions()
+
+    def _finish(self):
+        """Webview load finished."""
+        self._is_running = False
+        self.stop_spinner()
+        self.update_actions()
+
+    def _continue_initialization(self):
+        """Load home page."""
         self.go_home()
         QApplication.restoreOverrideCursor()
 
+    def _handle_url_combo_activation(self):
+        """Load URL from combo box first item."""
+        if not self._is_running:
+            text = str(self.url_combo.currentText())
+            self.go_to(self.text_to_url(text))
+        else:
+            self.get_action(WebViewActions.Stop).trigger()
+
+        self.get_focus_widget().setFocus()
+
+    def _change_url(self, url):
+        """
+        Displayed URL has changed -> updating URL combo box.
+        """
+        self.url_combo.add_text(self.url_to_text(url))
+
+    def _handle_icon_change(self):
+        """
+        Handle icone changes.
+        """
+        self.url_combo.setItemIcon(self.url_combo.currentIndex(),
+                                   self.webview.icon())
+        self.setWindowIcon(self.webview.icon())
+
+    # --- Qt overrides
+    # ------------------------------------------------------------------------
     def closeEvent(self, event):
-        """Handle close event"""
-        self.quit_server()
+        if self.is_server_running():
+            self.server.quit_server()
+
         event.accept()
 
-    # ------ Public API -------------------------------------------------------
+    # --- Public API
+    # ------------------------------------------------------------------------
+    def load_history(self, history):
+        """
+        Load history.
+
+        Parameters
+        ----------
+        history: list
+            List of searched items.
+        """
+        self.url_combo.addItems(history)
+
+    def initialize(self):
+        """Start pydoc server."""
+        QApplication.setOverrideCursor(QCursor(Qt.WaitCursor))
+        QApplication.processEvents()
+        self.start_server()
+        # Initializing continues in `_initialize_continued` method...

```

Comment:
This comment doesn't make sense to me. If it's not necessary, please remove it.

Reply:
🤷